File: ThreadAndroid/app/src/main/java/com/example/threadandroid/MainActivity.java
package com.example.threadandroid;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.os.MessageQueue;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Handler handler = new Handler(new Handler.Callback() {
            @Override
            public boolean handleMessage(@NonNull Message msg) {

                return false;
            }

        });
        handler.hasMessages(0);
        handler.getLooper();
        Looper.myQueue().addIdleHandler(new MessageQueue.IdleHandler() {
            @Override
            public boolean queueIdle() {
                return false;
            }
        });


        handler.obtainMessage(10).sendToTarget();
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            Handler.createAsync(Looper.myLooper());
        }

        Looper.prepareMainLooper();
    }

}

File: CustomViewBitmapAndDrawble/app/src/main/java/com/example/customviewbitmapanddrawble/MainActivity.kt
package com.example.customviewbitmapanddrawble

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

/**
 * Bitmap 和 Drawble
 * 什么是Bitmap
 * 什么是Drawable
 * Bitmap和Drawable互转的本质是什么
 * 自定义Drawable
 * 自定义Drawable的作用： 复用「绘制逻辑」
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /*val bitmap = Bitmap.createBitmap(20, 20, Bitmap.Config.ARGB_8888)
        bitmap.toDrawable(resources)
        val drawable = ColorDrawable(Color.RED)
        drawable.toBitmap()*/
    }
}